main/mySpace/session_filter.php

```diff
@@ -0,0 +1,267 @@
+                case ALL_DATE_FILTER:
+                    if ($createdAt >= $dateBegin && $createdAt <= $dateEnd) {
+                        $certificateList[] = $value;
+                    }
+                    break;
+            }
+        }
+    }
+}
+
+//select of sessions
+$sql = "SELECT id, name FROM $tblSession ORDER BY name";
+
+if (api_is_multiple_url_enabled()) {
+    $tblSessionRelAccessUrl = Database::get_main_table(TABLE_MAIN_ACCESS_URL_REL_SESSION);
+    $accessUrlId = api_get_current_access_url_id();
+    if ($accessUrlId != -1) {
+        $sql = "SELECT s.id, name FROM $tblSession s
+                INNER JOIN $tblSessionRelAccessUrl as session_rel_url
+                ON (s.id = session_rel_url.session_id)
+                WHERE access_url_id = $accessUrlId
+                ORDER BY name";
+    }
+}
+$result = Database::query($sql);
+$Sessions = Database::store_result($result);
+$options = [];
+$options['0'] = '';
+foreach ($Sessions as $enreg) {
+    $options[$enreg['id']] = $enreg['name'];
+}
+
+$form = new FormValidator('search_user', 'post', api_get_self());
+$form->addElement('select', 'session_id', get_lang('SessionList'), $options, ['id' => 'session-id']);
+$form->addDatePicker('date_begin', get_lang('DateStart'), ['id' => 'date-begin']);
+$form->addDatePicker('date_end', get_lang('DateEnd'), ['id' => 'date-end']);
+$form->addElement('hidden', 'formSent', 1);
+$form->addButtonSearch(get_lang('Search'));
+
+$interbreadcrumb[] = ['url' => 'index.php', 'name' => get_lang('MySpace')];
+Display::display_header(get_lang('CertificatesSessions'));
+echo Display::page_header(get_lang('CertificatesSessions'));
+$actions = '';
+$actions .= Display::url(
+    Display::return_icon('back.png', get_lang('Back'), [], 32),
+    api_get_path(WEB_CODE_PATH).'mySpace'
+);
+
+if ($allowCustomCertificate) {
+    $url = api_get_path(WEB_PLUGIN_PATH).'customcertificate/src/export_pdf_all_in_one.php';
+    $actions .= Display::url(
+        Display::return_icon('pdf.png', get_lang('ExportAllCertificatesToPDF'), [], ICON_SIZE_MEDIUM),
+        $url,
+        ['id' => 'export_pdf']
+    );
+
+    $actions .= Display::url(
+        Display::return_icon('file_zip.png', get_lang('ExportAllCertificatesToZIP'), [], ICON_SIZE_MEDIUM),
+        $url,
+        ['id' => 'export_zip']
+    );
+}
+
+echo Display::toolbarAction('actions', [$actions]);
+echo $form->returnForm();
+
+if (count($certificateList) == 0) {
+    echo Display::return_message(get_lang('NoResultsAvailable'), 'warning');
+} else {
+    echo '<table class="table data_table">';
+    echo '<tbody>';
+    foreach ($certificateList as $index => $value) {
+        $categoryId = $value['category_id'];
+        $courseCode = $value['course_code'];
+        $courseInfo = api_get_course_info($courseCode);
+        echo '<tr>';
+        echo '<td width="50%" class="actions">'.get_lang('Student').' : '.api_get_person_name($value['firstname'], $value['lastname']).' ('.$value['username'].')</td>';
```

Comment:
Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 168 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5744985)

Reply:
Corregir problemas indicados por codacy